src/enemy/gibdo.c

```diff
@@ -0,0 +1,434 @@
+    u8 x;
+    u32 x2;
+    if (this->bitfield == 0x87) {
+        if (this->action == 0x6) {
+            sub_08037ACC(this);
+        }
+        this->action = 0x8;
+        this->actionDelay = 0x3c;
+        this->flags &= 0x7F;
+        sub_08037B48(this);
+    } else {
+        if (this->action != 0x6) {
+            if (this->damageType == 0x27) {
+                if (this->bitfield == 0x80) {
+                    sub_08037A14(this);
+                }
+            } else {
+                if ((u8)(this->action - 1) < 2) {
+                    this->action = 1;
+                    x = ((this->field_0x3e ^ 0x10) + 4) & 0x18;
+                    this->direction = x;
+                    this->animationState = x >> 3;
+                    InitAnimationForceUpdate(this, this->animationState);
+                    if (sub_08037810(this) != 0) {
+                        this->actionDelay = 4;
+                    }
+                }
+            }
+        } else {
+            sub_08037A58(this);
+        }
+    }
+    if (this->field_0x43 != 0) {
+        sub_0804A9FC(this, 0x1c);
+    }
+    sub_0804AA30(this, gGibdo);
+}
+END_NONMATCH
+
+void nullsub_162() {
+}
+
+void sub_08037558(Entity* this) {
+    u32 r;
+    sub_0804A720(this);
+    r = Random();
+    this->animationState = r & 0x3;
+    this->field_0x76.HALF.LO = 0;
+    this->field_0x76.HALF.HI = 0;
+    sub_08037794(this);
+}
+
+void sub_08037580(Entity* this) {
+    if (sub_08037810(this) == 0) {
+        this->field_0x74.HWORD--;
+        if (this->field_0x74.HWORD == 0) {
+            sub_080377B0(this);
+        }
+    }
+}
+void sub_080375A4(Entity* this) {
+    if (sub_080378B0(this) == 0) {
+        if (sub_08037810(this) == 0) {
+            this->field_0x74.HWORD--;
+            if (this->field_0x74.HWORD == 0) {
+                sub_08037794(this);
+            } else {
+                UpdateAnimationSingleFrame(this);
+                if (ProcessMovement(this) == 0) {
+                    this->field_0xf--;
+                    if (this->field_0xf == 0) {
+                        sub_080379BC(this);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void sub_080375F8(Entity* this) {
+    if (sub_080378B0(this) == 0) {
+        this->actionDelay--;
+        if (this->actionDelay == 0) {
+            this->action = 4;
+            this->actionDelay = 0x18;
+            InitAnimationForceUpdate(this, this->animationState + 4);
+        }
+    }
+}
+
+void sub_08037624(Entity* this) {
+    if (sub_080378B0(this) == 0) {
+        if (sub_08037914(this) == 0) {
+            this->field_0x74.HWORD--;
+            if (this->field_0x74.HWORD == 0) {
+                sub_08037794(this);
+            } else {
+                UpdateAnimationSingleFrame(this);
+                UpdateAnimationSingleFrame(this);
+                if (ProcessMovement(this) == 0) {
+                    this->field_0xf--;
+                    if (this->field_0xf == 0) {
```

Comment:
these and similar ones can usually be written as simply `if (--this->field_0xf)`